### ISSUE TITLE
python: private method `xz_dep' called

### ISSUE THREAD

**Issue by ghost** [Linuxbrew/brew#184]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

I'm using `brew install node` or `brew install python` on ubuntu in Docker (as always some weeks ago).

Today I run into this:

```
Step 28 : RUN brew install python
 ---> Running in e3ea152b8525
Updating Homebrew...
==> Tapping homebrew/dupes
Cloning into '/home/john/.linuxbrew/Library/Taps/homebrew/homebrew-dupes'...
Tapped 38 formulae (105 files, 337.0K)
==> Installing dependencies for python: patchelf, xz, pkg-config, homebrew/dupes/ncurses, readline, sqlite, gdbm, makedepend, zlib, openssl, bzip2, gpatch
==> Installing python dependency: patchelf
==> Downloading https://linuxbrew.bintray.com/bottles/patchelf-0.9_1.x86_64_linux.bottle.tar.gz
==> Pouring patchelf-0.9_1.x86_64_linux.bottle.tar.gz
🍺  /home/john/.linuxbrew/Cellar/patchelf/0.9_1: 6 files, 1.2M
==> Installing python dependency: xz
==> Downloading https://linuxbrew.bintray.com/bottles/xz-5.2.2.x86_64_linux.bottle.tar.gz
==> Pouring xz-5.2.2.x86_64_linux.bottle.tar.gz
🍺  /home/john/.linuxbrew/Cellar/xz/5.2.2: 98 files, 1.8M
==> Installing python dependency: pkg-config
==> Downloading https://pkgconfig.freedesktop.org/releases/pkg-config-0.29.1.tar.gz
==> ./configure --prefix=/home/john/.linuxbrew/Cellar/pkg-config/0.29.1_2 --disable-host-tool --with-internal-glib --with-pc-path=/home/john/.linuxbrew/lib/pkgconfig:/home/john/.linuxbrew/share/pkgconfig:/usr/local/lib/pkgconfig:/usr/lib/pkgconfig:/home/john/.linuxbrew/Library/Homebrew/os/mac/pkgconfig/0
==> make
==> make check
==> make install
🍺  /home/john/.linuxbrew/Cellar/pkg-config/0.29.1_2: 11 files, 677.8K, built in 38 seconds
==> Installing python dependency: homebrew/dupes/ncurses
==> Downloading https://ftpmirror.gnu.org/ncurses/ncurses-6.0.tar.gz
==> Downloading from http://mirror2.mirror.garr.it/mirrors/gnuftp/gnu/ncurses/ncurses-6.0.tar.gz
==> ./configure --prefix=/home/john/.linuxbrew/Cellar/ncurses/6.0_2 --enable-pc-files --with-pkg-config-libdir=/home/john/.linuxbrew/Cellar/ncurses/6.0_2/lib/pkgconfig --enable-sigwinch --enable-symlinks --enable-widec --mandir=/home/john/.linuxbrew/Cellar/ncurses/6.0_2/share/man --with-manpage-format=normal --with-shared --with-gpm=no
==> make
==> make install
🍺  /home/john/.linuxbrew/Cellar/ncurses/6.0_2: 3,831 files, 14.9M, built in 1 minute 11 seconds
==> Installing python dependency: readline
==> Downloading https://linuxbrew.bintray.com/bottles/readline-7.0.x86_64_linux.bottle.tar.gz
==> Pouring readline-7.0.x86_64_linux.bottle.tar.gz
🍺  /home/john/.linuxbrew/Cellar/readline/7.0: 46 files, 1.7M
==> Installing python dependency: sqlite
==> Downloading https://linuxbrew.bintray.com/bottles/sqlite-3.15.0.x86_64_linux.bottle.tar.gz
==> Pouring sqlite-3.15.0.x86_64_linux.bottle.tar.gz
🍺  /home/john/.linuxbrew/Cellar/sqlite/3.15.0: 12 files, 2.8M
==> Installing python dependency: gdbm
==> Downloading https://linuxbrew.bintray.com/bottles/gdbm-1.12.x86_64_linux.bottle.tar.gz
==> Pouring gdbm-1.12.x86_64_linux.bottle.tar.gz
🍺  /home/john/.linuxbrew/Cellar/gdbm/1.12: 29 files, 806.4K
==> Installing python dependency: makedepend
==> Downloading https://linuxbrew.bintray.com/bottles/makedepend-1.0.5.x86_64_linux.bottle.1.tar.gz
==> Pouring makedepend-1.0.5.x86_64_linux.bottle.1.tar.gz
🍺  /home/john/.linuxbrew/Cellar/makedepend/1.0.5: 7 files, 96.4K
==> Installing python dependency: zlib
==> Downloading https://linuxbrew.bintray.com/bottles/zlib-1.2.8.x86_64_linux.bottle.1.tar.gz
==> Pouring zlib-1.2.8.x86_64_linux.bottle.1.tar.gz
🍺  /home/john/.linuxbrew/Cellar/zlib/1.2.8: 11 files, 398.6K
==> Installing python dependency: openssl
==> Downloading https://www.openssl.org/source/openssl-1.0.2j.tar.gz
==> perl ./Configure --prefix=/home/john/.linuxbrew/Cellar/openssl/1.0.2j --openssldir=/home/john/.linuxbrew/etc/openssl no-ssl2 zlib-dynamic shared enable-cms -isystem/home/john/.linuxbrew/include -Os -w -pipe -march=native -L/home/john/.linuxbrew/lib -Wl,--dynamic-linker=/home/john/.linuxbrew/lib/ld.so -Wl,-rpath,/home/john/.linuxbrew/lib linux-x86_64
==> make depend
==> make
==> make test
==> make install MANDIR=/home/john/.linuxbrew/Cellar/openssl/1.0.2j/share/man MANSUFFIX=ssl
==> Downloading https://curl.haxx.se/ca/cacert-2016-04-20.pem
==> Caveats
A CA file has been bootstrapped using certificates from the SystemRoots
keychain. To add additional certificates (e.g. the certificates added in
the System keychain), place .pem files in
  /home/john/.linuxbrew/etc/openssl/certs

and run
  /home/john/.linuxbrew/opt/openssl/bin/c_rehash

==> Summary
🍺  /home/john/.linuxbrew/Cellar/openssl/1.0.2j: 1,703 files, 14.6M, built in 2 minutes 33 seconds
==> Installing python dependency: bzip2
==> Downloading https://linuxbrew.bintray.com/bottles/bzip2-1.0.6_1.x86_64_linux.bottle.1.tar.gz
==> Pouring bzip2-1.0.6_1.x86_64_linux.bottle.1.tar.gz
🍺  /home/john/.linuxbrew/Cellar/bzip2/1.0.6_1: 29 files, 1.5M
==> Installing python dependency: gpatch
==> Downloading https://linuxbrew.bintray.com/bottles/gpatch-2.7.5.x86_64_linux.bottle.1.tar.gz
==> Pouring gpatch-2.7.5.x86_64_linux.bottle.1.tar.gz
🍺  /home/john/.linuxbrew/Cellar/gpatch/2.7.5: 9 files, 313.6K
==> Installing python
==> Downloading https://www.python.org/ftp/python/2.7.12/Python-2.7.12.tar.xz
Error: private method `xz_dep' called for #<DependencyCollector:0x00000001ca9670>
Please report this bug:
  https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
/home/john/.linuxbrew/Library/Homebrew/dependency_collector.rb:65:in `tar_needs_xz_dependency?'
/home/john/.linuxbrew/Library/Homebrew/download_strategy.rb:244:in `block in stage'
/home/john/.linuxbrew/Library/Homebrew/utils.rb:255:in `with_system_path'
/home/john/.linuxbrew/Library/Homebrew/download_strategy.rb:243:in `stage'
/home/john/.linuxbrew/Library/Homebrew/resource.rb:95:in `block in unpack'
/home/john/.linuxbrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/home/john/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/home/john/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/home/john/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/home/john/.linuxbrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/home/john/.linuxbrew/Library/Homebrew/resource.rb:94:in `unpack'
/home/john/.linuxbrew/Library/Homebrew/resource.rb:86:in `stage'
/home/john/.linuxbrew/Library/Homebrew/formula.rb:1799:in `stage'
/home/john/.linuxbrew/Library/Homebrew/formula.rb:1075:in `brew'
/home/john/.linuxbrew/Library/Homebrew/build.rb:112:in `install'
/home/john/.linuxbrew/Library/Homebrew/build.rb:189:in `<main>'
The command '/bin/sh -c brew install python' returned a non-zero code: 1
```

*Why???*

**Comment by bw-yuhogg** [comment]
Thanks for reporting @johnunclesam!

Can you do another brew update and try again please? I believe I fixed this issue earlier this morning (#183).

**Comment by bw-yuhogg** [comment]
EDIT: my apologies; I forgot that #183 hasn't gotten into a stable release yet. I'll try to make one as soon as possible.

**Comment by bw-yuhogg** [comment]
@johnunclesam, I've released a new stable build that includes #183. Please retry the installation; you should be updated to the fixed build automatically.

Thanks a bunch and my apologies 🐱

**Comment by sjackman** [comment]
This issue should be resolved by Linuxbrew 1.1.1.
